analyses/cms-open-data-ttbar/utils.py

```diff
@@ -1,6 +1,34 @@
-from tqdm import tqdm
-from urllib.request import urlretrieve
+import json
+from dataclasses import dataclass
 from pathlib import Path
+from typing import Optional, Union
+from urllib.request import urlretrieve
+
+import ROOT
+from tqdm import tqdm
+
+
+@dataclass
+class AGCInput:
+    paths: list[str]  # paths, http urls or xrootd urls of the input files
```

Comment:
Usage of `list[str]` sets the minimum Python version to 3.9. I think it's fine, but should be documented somewhere

Reply:
https://github.com/root-project/analysis-grand-challenge/pull/20 ?